scanpy/plotting/_anndata.py

```diff
@@ -996,10 +996,17 @@ def heatmap(
         dendrogram = False
     else:
         categorical = True
-        # get categories colors:
-        if groupby + "_colors" in adata.uns:
+        # get categories colors
+        if is_categorical_dtype(adata.obs[groupby]):
+            if groupby + "_colors" not in adata.uns:
```

Comment:
`groupby` is documented as being able to be `Union[str, Sequence[str]]`. Is that actually possible here?

Reply:
I added a comment for this issue. If groupby is a sequence then the colors are not saved. This can be fixed but requires further changes that at the moment I don't have the time to make.